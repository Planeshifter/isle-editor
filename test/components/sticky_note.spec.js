// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import StickyNote from 'components/sticky-note/main.js';


// TESTS //

it( 'renders a sticky note', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders an editable sticky note', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" editable />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders a minimizable sticky note', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" minimizable />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders a colored sticky note', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" color="orange" />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders a sticky note with a date', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" date="01/20/2020" />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders a sticky note with a stain', () => {
	const calculator = renderer.create(
		<StickyNote title="My note" body="Text body of note" stain />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});
