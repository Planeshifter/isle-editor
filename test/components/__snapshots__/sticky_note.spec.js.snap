// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a colored sticky note 1`] = `
<div
  className="sticky-note-outer"
  style={
    Object {
      "filter": "hue-rotate(320deg) saturate(70%)",
      "webkitFilter": "hue-rotate(320deg) saturate(70%)",
    }
  }
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      />
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title"
        >
          My note
        </div>
        <div
          className="sticky-note-date"
        >
          
        </div>
        <div
          className="sticky-note-body"
        >
          Text body of note
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a minimizable sticky note 1`] = `
<div
  className="sticky-note-outer"
  style={Object {}}
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      >
        <div
          className="sticky-note-minimizable"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          â€“
        </div>
      </div>
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title"
        >
          My note
        </div>
        <div
          className="sticky-note-date"
        >
          
        </div>
        <div
          className="sticky-note-body"
        >
          Text body of note
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a sticky note 1`] = `
<div
  className="sticky-note-outer"
  style={Object {}}
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      />
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title"
        >
          My note
        </div>
        <div
          className="sticky-note-date"
        >
          
        </div>
        <div
          className="sticky-note-body"
        >
          Text body of note
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a sticky note with a date 1`] = `
<div
  className="sticky-note-outer"
  style={
    Object {
      "filter": "hue-rotate(320deg) saturate(70%)",
      "webkitFilter": "hue-rotate(320deg) saturate(70%)",
    }
  }
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      />
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title"
        >
          My note
        </div>
        <div
          className="sticky-note-date"
        >
          01/20/2020
        </div>
        <div
          className="sticky-note-body"
        >
          Text body of note
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a sticky note with a stain 1`] = `
<div
  className="sticky-note-outer"
  style={
    Object {
      "filter": "hue-rotate(320deg) saturate(70%)",
      "webkitFilter": "hue-rotate(320deg) saturate(70%)",
    }
  }
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      />
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title"
        >
          My note
        </div>
        <div
          className="sticky-note-date"
        >
          
        </div>
        <div
          className="sticky-note-body"
        >
          Text body of note
        </div>
        <div
          className="sticky-note-stain"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an editable sticky note 1`] = `
<div
  className="sticky-note-outer"
  style={Object {}}
>
  <div
    className="sticky-note"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="sticky-note-wrapper"
    >
      <div
        className="sticky-note-controls"
      />
      <div
        className="sticky-note-content"
      >
        <div
          className="sticky-note-title editable"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          My note
        </div>
        <div
          className="sticky-note-body editable"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          Text body of note
        </div>
      </div>
    </div>
  </div>
</div>
`;
